UQ-eLIPSE/prism-server
# feature/TEAM24-159

**Setup:**
1. `git checkout feature/TEAM24-159`
2. `cd client` and run `yarn` to add any missing dependecies.
3. When changing sites to test, change the `DATABASE_URL` to your specific site and restart server for docker.
i.e. `mongodb://mongodb:27017/aeb` to `mongodb://mongodb:27017/kingston` to test kingston.
4. run `localhost:3000/site`.

**Testing:**
1. Move the marzipano camera viewer anywhere and note the point of view of the selected node in the minimap.
2. Try moving to another floor, the selected node's POV should remain consistent. 
6. Now click on another node on the same floor, the POV should be resetted to that specific node's configuration setting.
7. Run `yarn cypress:run:local` or read the README based on the current active project you're on. i.e. `yarn cypress:run:local-agco360` for agco360 site. Please do not test `agco360` and `uwmt` sites.
8. Currently, one of the rotation tests in the `MinimapTest` may fail but I've discussed this @Jo-Zh and @Mohamad11Dab and we talked about how the future rotation state can be refactored. It was decided this functionality can be implemented and the test or rotation concept fixed at a later date.
